addons/Magic_Boxes/src/presenter.js

```diff
@@ -1264,6 +1326,91 @@ function AddonMagic_Boxes_create() {
         presenter.isSelectionPossible = true;
 
         presenter.isShowAnswersActive = false;
+        presenter.isGradualShowAnswersActive = false;
+    };
+
+    presenter.gradualShowAnswers = function () {
+        if(!presenter.isGradualShowAnswersActive){
+            presenter.saveUserSelectionBeforeShowAnswers();
+        }
+
+        presenter.isShowAnswersActive = true;
+        presenter.isGradualShowAnswersActive = true;
+        presenter.isSelectionPossible = false;
+        
+
+        if (presenter.configuration.checkByWords){
+            presenter.showNextWord();
+        } else {
+            presenter.showNextLetter();
+        }
+    };
+
+    presenter.showNextWord = function () {
+        var columns = presenter.configuration.columns;
+
+        for (var key in presenter.GSAcorrectAnswerLocations){
+            if (usedGSAKeys.includes(key)) {
+                continue;
+            } else {
+                usedGSAKeys.push(key);
+            }
+            if (!presenter.GSAcorrectAnswerLocations.hasOwnProperty(key)) continue;
+            var answer = presenter.GSAcorrectAnswerLocations[key];
+
+            var row = answer["row"];
+            var column = answer["column"];
+
+            if (row.length != column.length) continue;
+            for (let i = 0; i < row.length; i++){
+                let index = row[i] * columns + column[i];
+                var element = gridContainerWrapper.find(".selectable-element:eq(" + index + ")");
+                element.addClass('selectable-element-show-answers');
+            }
+            break;
+        }
+    };
+
+    presenter.showNextLetter = function () {
+        var columns = presenter.configuration.columns;
+
+        for (var key in presenter.GSAcorrectAnswerLocations){
+            if (usedGSAKeys.includes(key)) {
+                continue;
+            }
+            if (!presenter.GSAcorrectAnswerLocations.hasOwnProperty(key)) continue;
+            var answer = presenter.GSAcorrectAnswerLocations[key];
+
+            var row = answer["row"];
+            var column = answer["column"];
+
+            if (row.length != column.length) continue;
+            
+            var element;
+            do {
+                let index = row[GSALetterCounter] * columns + column[GSALetterCounter];
+                element = gridContainerWrapper.find(".selectable-element:eq(" + index + ")");
```

Comment:
To można zamienić na interpolacje stringa 
```suggestion
                element = gridContainerWrapper.find(`.selectable-element:eq(${index}));
```